src/wp-includes/link-template.php

```diff
@@ -1597,25 +1597,39 @@ function get_delete_post_link( $post = 0, $deprecated = '', $force_delete = fals
  * @since 2.3.0
  *
  * @param int|WP_Comment $comment_id Optional. Comment ID or WP_Comment object.
+ * @param string         $context    Optional. Context in which the URL should be used. Either 'display',
+ *                                   to include HTML entities, or 'display'. Default 'display'.
  * @return string|void The edit comment link URL for the given comment.
  */
-function get_edit_comment_link( $comment_id = 0 ) {
+function get_edit_comment_link( $comment_id = 0, $context = 'display' ) {
 	$comment = get_comment( $comment_id );
 
+	if ( ! ( is_object( $comment ) && property_exists( $comment, 'comment_ID' ) ) ) {
+		return null;
+	}
+
 	if ( ! current_user_can( 'edit_comment', $comment->comment_ID ) ) {
 		return;
 	}
 
-	$location = admin_url( 'comment.php?action=editcomment&amp;c=' ) . $comment->comment_ID;
+	if ( 'display' === $context ) {
+		$action = 'comment.php?action=editcomment&amp;c=';
+	} else {
+		$action = 'comment.php?action=editcomment&c=';
+	}
+
+	$location = admin_url( $action ) . $comment->comment_ID;
 
 	/**
 	 * Filters the comment edit link.
 	 *
 	 * @since 2.3.0
```

Comment:
Similar to above, since you're adding new parameters, this change should be documented here.
```suggestion
	 * @since 2.3.0
	 * @since 6.7.0 The $comment_id and $context parameters are now being passed to the filter.
```